scene/gui/text_edit.h

```diff
@@ -47,6 +47,44 @@ class TextEdit : public Control {
 		GUTTER_TPYE_CUSTOM
 	};
 
+	struct Selection {
+		enum Mode {
```

Comment:
Might be worth moving this out so we don't have to expose the entire struct, renaming it to `SelectionMode`